### ISSUE TITLE
Can't access `mcmcP` output because `MHResult` is in a hidden module

### ISSUE THREAD

**Issue by rihardsk** [tweag/monad-bayes#315]
Not exactly a bug, but probably an oversight?

Sampling with [mcmcP](https://github.com/tweag/monad-bayes/blob/371d05cfd002a1916d80c1ff092b2c5b6e275adf/src/Control/Monad/Bayes/Inference/MCMC.hs#L53C1-L53C1) produces samples wrapped in `MHResult`, but I can't access anything because `MHResult` (and `Trace` that's contained within) is defined in `Control.Monad.Bayes.Traced.Common` and `monad-bayes` doesn't export that module (there are no re-exports in other modules either).

**Comment by turion** [comment]
True. This should be taken into account when fixing https://github.com/tweag/monad-bayes/issues/260.
